src/DocBlock/TypeExpression.php

```diff
@@ -252,11 +253,22 @@ public function getTypes(): array
         return [$this->value];
     }
 
+    public function isGluedType(): bool
+    {
+        return $this->isGluedType;
+    }
+
+    /**
+     * @deprecated Use better named self::isGluedType() method instead
+     */
     public function isUnionType(): bool
     {
-        return $this->isUnionType;
+        return $this->isGluedType();
```

Comment:
Instead of deprecating this we should improve the condition and allow determining real union types:

```suggestion
        return $this->isCompositeType && '|' === $this->typesGlue;
```

we also should add `isIntersectionType()`.

Reply:
reverted completely because of `~` not implemented in this PR - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210#discussion_r1765400782

Reply:
reverted completely because of `~` not implemented in this PR - https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8210#discussion_r1765400782